network/p2p/libp2pNode.go

```diff
@@ -283,6 +288,31 @@ func (n *Node) WithDefaultUnicastProtocol(defaultHandler libp2pnet.StreamHandler
 	return nil
 }
 
+func (n *Node) WithPeersProvider(peersProvider PeersProvider) error {
+	if n.peerManagerFactory == nil {
+		return nil
```

Comment:
not an error case?

Reply:
No, it's an expected case. In the previous implementation, not all middleware instances had a `peerManagerFactory`. This replicates that behavior.

That's currently used by observers/consensus followers, which don't use a peer manager.